### PR TITLE
fix: address copilot review feedback on triage sidebar

### PR DESCRIPTION
**Pull request by joshdholtz**
## Summary

- **Harden `parse_github_label`**: strip query string/fragment, validate that path segment 5 is `issues` or `pull` (not `tree`, `blob`, etc.), validate number segment is numeric
- **Add unit tests** for `parse_github_label`: issue URL, PR URL, trailing slashes, query/fragment stripping, non-GitHub URL → None, non-issue/PR path → None
- **Fix Unicode panic in `truncate_to_width`**: replace byte-slice indexing (`&s[..n]`) with char-based `chars().count()` / `chars().take(n)` — prevents panics on emoji and CJK in GitHub issue titles
- **Remove 3 background-only `Paragraph::new("")` widgets per item**: apply `row_bg` style directly to the content `Paragraph` widgets (line1/line2/line3), cutting render work by ~50% per item
- **Reserve age width on Line 2**: compute `age_suffix` width first, then truncate `source_label` to leave room — long labels no longer push the age off-screen

## Test plan
- [ ] `cargo test -p apiari` — all 720 tests pass including 7 new `parse_github_label` tests
- [ ] `cargo clippy -p apiari -- -D warnings` — clean
- [ ] `cargo fmt -p apiari` — no changes

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Review by joshdholtz — COMMENTED** [review]
## Review: LGTM ✓

Verified locally — 720 tests pass (7 new), clippy clean, fmt clean.

### Changes look correct:

**`parse_github_label` hardening** (`app.rs:456–475`) — query/fragment stripping via `split('?').next()` / `split('#').next()`, `issues`/`pull` kind validation, numeric segment check. All 7 edge-case tests cover the right paths.

**Unicode panic fix** (`render.rs:872–883`) — `s.chars().take(n)` replaces `&s[..n]` byte-slice indexing. Correct fix; `format_age` produces only ASCII so no double-width character issues in the age width reservation either.

**Background widget removal** (`render.rs:753–762`) — applying `row_bg` directly to content `Paragraph` widgets is semantically equivalent and reduces render work.

**Age width reservation** (`render.rs:794–802`) — `label_max` correctly subtracts prefix (3 chars) and `age_suffix.chars().count()` with `saturating_sub` to prevent underflow on narrow terminals.

No issues found. Ready to merge.

File: crates/apiari/src/ui/render.rs [hunk 1]
```diff
@@ -879,10 +872,12 @@ fn format_age(dur: &chrono::Duration) -> String {
...
+    let char_count = s.chars().count();
+    if char_count <= max_width {
         s.to_string()
     } else if max_width > 1 {
-        format!("{}…", &s[..max_width - 1])
+        let truncated: String = s.chars().take(max_width - 1).collect();
+        format!("{truncated}…")
     } else {
         "…".to_string()
     }
```
[the remark this answers is not in the document]
Comment (joshdholtz) [reply crates/apiari/src/ui/render.rs:894]: Fixed in the follow-up commit. `truncate_to_width` now accumulates display columns using `UnicodeWidthChar::width` rather than counting chars, and cuts at the correct byte boundary before appending '…'. CJK characters (width 2) and emoji (width 1 or 2) are now handled correctly.

File: crates/apiari/src/ui/app.rs [hunk 2]
```diff
@@ -456,12 +456,20 @@ pub struct TriageItem {
...
     let parts: Vec<&str> = url.split('/').collect();
+    // Expected shape: ["https:", "", "github.com", "owner", "repo", "issues"|"pull", "123"]
     if parts.len() >= 7 && parts[2] == "github.com" {
+        let kind = parts[5];
         let repo = parts[4];
         let number = parts[6];
-        return Some(format!("{repo} #{number}"));
+        if (kind == "issues" || kind == "pull") && number.chars().all(|c| c.is_ascii_digit()) {
+            return Some(format!("{repo} #{number}"));
+        }
```
[the remark this answers is not in the document]
Comment (joshdholtz) [reply crates/apiari/src/ui/app.rs:478]: Fixed. Added `!owner.is_empty() && !repo.is_empty()` guards before accepting the URL. Also added a unit test: `test_parse_github_label_empty_repo_segment_returns_none` that asserts `https://github.com/owner//issues/123` returns `None`.

File: crates/apiari/src/ui/render.rs [hunk 3]
```diff
@@ -802,25 +791,29 @@ fn draw_triage_sidebar(frame: &mut Frame, ws: &app::WorkspaceState, area: Rect)
...
             y: inner.y + y_offset + 1,
             width: inner.width,
             height: 1,
         };
+        // prefix "▌  " = 3 chars; suffix " · {age}" reserves separator + age
+        let age_suffix = format!(" \u{b7} {age_str}");
+        let label_max = (inner.width as usize)
+            .saturating_sub(3) // bar + two spaces
+            .saturating_sub(age_suffix.chars().count());
+        let truncated_label = truncate_to_width(&item.source_label, label_max);
```
[the remark this answers is not in the document]
Comment (joshdholtz) [reply crates/apiari/src/ui/render.rs]: Fixed. `age_suffix_width` is now computed by summing `UnicodeWidthChar::width` over each char in `age_suffix`, giving the true terminal column count. `label_max` subtracts this display width rather than a char count.

File: crates/apiari/src/ui/render.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (joshdholtz) [reply crates/apiari/src/ui/render.rs:894]: Addressed in commit c46bb43. `truncate_to_width` now uses `UnicodeWidthStr::width(s)` for the initial check and iterates via `char_indices()` to accumulate display columns with `UnicodeWidthChar::width(ch)`, cutting at the exact byte boundary when the next character would exceed the budget. The ellipsis occupies 1 column of the reserved space.

File: crates/apiari/src/ui/app.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (joshdholtz) [reply crates/apiari/src/ui/app.rs:478]: Already fixed in 7df21b5: `!owner.is_empty() && !repo.is_empty()` guards are in place, plus a `test_parse_github_label_empty_repo_segment_returns_none` test confirming `https://github.com/owner//issues/123` → `None`. Thread resolved.

File: crates/apiari/src/ui/render.rs (hunk 3 above)
[the remark this answers is not in the document]
Comment (joshdholtz) [reply crates/apiari/src/ui/render.rs]: Addressed in commit c46bb43. `label_max` now subtracts `UnicodeWidthStr::width(age_suffix.as_str())`, giving the exact terminal column count for the suffix rather than a char count.

File: crates/apiari/src/ui/render.rs [hunk 4]
```diff
@@ -879,13 +872,24 @@ fn format_age(dur: &chrono::Duration) -> String {
...
-    if s.len() <= max_width {
-        s.to_string()
-    } else if max_width > 1 {
-        format!("{}…", &s[..max_width - 1])
-    } else {
-        "…".to_string()
+    if UnicodeWidthStr::width(s) <= max_width {
+        return s.to_string();
+    }
+    if max_width <= 1 {
```
[the remark this answers is not in the document]
Comment (joshdholtz) [reply crates/apiari/src/ui/render.rs]: Fixed in e71bb18. Added an explicit `max_width == 0` guard that returns `String::new()` before the `max_width == 1` ellipsis case, so a zero-column budget produces no output rather than a 1-column '…' that overflows the Rect.
